### PR TITLE
Adding Log Lines, Step Details, and Permutations

### PR DESCRIPTION
**Pull request by stjohnjohnson**
- Starting a new Log Lines model
- Adding step details to Build
- Adding Permutations to Job

### CODE REVIEW

File: models/log.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    base: Joi.object(MODEL).label('Log'),
+
+    /**
+     * Properties for Log that will come back during a GET request
+     *
+     * @property get
+     * @type {Joi}
+     */
+    get: Joi.object(mutate(MODEL, [
+        'id', 'buildId', 'timestamp', 'message', 'number', 'stepName', 'time'
```
Comment (minz1027) [inline models/log.js]: Why 'timestamp' is here? Only 'time' is defined in the MODEL.

Comment (minz1027) [inline models/log.js]: Why 'timestamp' is here? Only 'time' is defined in the MODEL.

File: models/job.js [hunk 2]
```diff
@@ -68,8 +71,7 @@ module.exports = {
      * @type {Joi}
      */
     update: Joi.object(mutate(MODEL, [], [
-        'state',
-        'containers'
+        'state'
```
Comment (d2lam) [inline models/job.js]: we don't allow update of containers anymore?

Comment (stjohnjohnson) [inline models/job.js]: We don't allow update of builds other than status.

Comment (nkatzman) [inline models/job.js]: This is for jobs?

Should containers only ever be updated through the screwdriver.yaml?

Comment (stjohnjohnson) [inline models/job.js]: Correct.

File: models/log.js [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+     */
+    tableName: 'logs',
+
+    /**
+     * List of indexes to create in the datastore
+     *
+     * @property indexes
+     * @type {Array}
+     */
+    indexes: ['buildId', 'stepName']
```
Comment (minz1027) [inline models/log.js]: In the doc, the Log API will `get step#1 logs from lineNumber` to get logs from db. If we only index `buildId` and `stepName`, does it mean we need to scan all lines for the specified step to find the lines we want?

Comment (stjohnjohnson) [inline models/log.js]: Good point, I'm really not sure what happens with such a large number of indexes.  @nkatzman

**Comment by minz1027** [comment]
LGTM 👍

File: test/data/build.yaml [hunk 4]
```diff
@@ -14,3 +14,20 @@ parameters:
 meta:
     coverage: 99.54
 status: SUCCESS
```
Comment (petey) [inline test/data/build.yaml]: This really doesn't look like a successful job based on the status codes in steps.

Comment (stjohnjohnson) [inline test/data/build.yaml]: Heh

**Comment by petey** [comment]
👍  
stjohn aliens

**Comment by d2lam** [comment]
👍

**Pull request merged**